src/main/java/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        for (Map.Entry<String, Integer> fruit : Storage.fruitsStorage.entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                fruit.setValue(Math.max(fruit.getValue() - fruitTransaction.getQuantity(), 0));
+                return;
+            }
+        }
+    }
```

Comment:
According to the checklist, handling the Purchase operation should include a check for a positive balance after the operation. If the balance is negative, a RuntimeException should be thrown. Please ensure that the operation method checks the result balance before saving it in the Storage and throws an exception if the balance would be negative.